acl_loader/main.py

```diff
@@ -0,0 +1,569 @@
+                         val.get("ttl", ""), val.get("queue", ""), val.get("tos", "")])
+
+        print(tabulate.tabulate(data, headers=header, tablefmt="simple", missingval=""))
+
+    def show_rule(self, table_name, rule_id):
+        """
+        Show ACL rules configuration.
+        :param table_name: Optional. ACL table name. Filter rules by specified table name.
+        :param rule_id: Optional. ACL rule name. Filter rule by specified rule name.
+        :return:
+        """
+        header = ("Rule ID", "Table Name", "Priority", "Action", "Match")
+
+        ignore_list = ["PRIORITY", "PACKET_ACTION", "MIRROR_ACTION"]
+
+        raw_data = []
+        for (tname, rid), val in self.get_rules_db_info().iteritems():
+
+            if table_name and table_name != tname:
+                continue
+
+            if rule_id and rule_id != rid:
+                continue
+
+            priority = val["PRIORITY"]
+
+            action = ""
+            if "PACKET_ACTION" in val:
+                action = val["PACKET_ACTION"]
+            elif "MIRROR_ACTION" in val:
+                action = "MIRROR: %s" % val["MIRROR_ACTION"]
+            else:
+                continue
+
+            matches = ["%s: %s" % (k, v) for k, v in val.iteritems() if k not in ignore_list]
+
+            matches.sort()
+
+            rule_data = [[tname, rid, priority, action, matches[0]]]
+            if len(matches) > 1:
+                for m in matches[1:]:
+                    rule_data.append(["", "", "", "", m])
+
+            raw_data.append([priority, rule_data])
+
+        def cmp_rules(a, b):
+            return cmp(a[0], b[0])
+
+        raw_data.sort(cmp_rules)
+        raw_data.reverse()
+
+        data = []
+        for _, d in raw_data:
+            data += d
+
+        print(tabulate.tabulate(data, headers=header, tablefmt="simple", missingval=""))
+
+
+@click.group()
+@click.pass_context
+def cli(ctx):
+    """
+    Utility entry point.
+    """
+    context = {
+        "acl_loader": AclLoader()
+    }
+
+    ctx.obj = context
+
+
+@cli.group()
+@click.pass_context
+def show(ctx):
+    """
+    Show ACL configuration.
+    """
+    pass
+
+
+@show.command()
+@click.argument('table_name', type=click.STRING, required=False)
+@click.pass_context
+def table(ctx, table_name):
+    """
+    Show ACL tables configuration.
+    :return:
+    """
+    acl_loader = ctx.obj["acl_loader"]
+    acl_loader.show_table(table_name)
+
+
+@show.command()
+@click.argument('session_name', type=click.STRING, required=False)
+@click.pass_context
+def session(ctx, session_name):
+    """
+    Show mirror session configuration.
+    :return:
+    """
+    acl_loader = ctx.obj["acl_loader"]
+    acl_loader.show_session(session_name)
+
+
+@show.command()
+@click.argument('table_name', type=click.STRING, required=False)
+@click.argument('rule_id', type=click.STRING, required=False)
+@click.pass_context
+def rule(ctx, table_name, rule_id):
```

Comment:
Can we add an option to specify max_priority explicitly instead of using default value?

Reply:
Do you want to make it mandatory? Or we still should use a default value if priority is not provided by user?

Reply:
It should be optional and we should fall back to default value if not specified by user.

Reply:
done